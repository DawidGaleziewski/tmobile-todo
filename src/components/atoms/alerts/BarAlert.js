import React from "react";

// Libraries
import { jsx, css } from "@emotion/core";

// Styles
import { alertStyle, closeBtnStyle } from "./style/BarAlertStyles";

/**@jsx jsx */

const BarAlert = ({ alertText, alertType, removeAlert, id }) => {
  const closeBtnClickHandler = (event) => {
    removeAlert(id);
  };
  return (
    <div css={alertStyle(alertType)} data-testid="bar-alert">
      {alertText}
      <span onClick={closeBtnClickHandler} css={closeBtnStyle}>
        X
      </span>
    </div>
  );
};

export default BarAlert;
