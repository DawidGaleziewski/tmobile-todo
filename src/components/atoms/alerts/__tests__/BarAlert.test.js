// Libraries
import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

// Components
import BarAlert from "../BarAlert";

afterEach(cleanup);

it("renders BarAlert without crashing", () => {
  const divContainer = document.createElement("div");
  ReactDOM.render(<BarAlert />, divContainer);
  ReactDOM.unmountComponentAtNode(divContainer);
});

it("renders BarAlert correctly", () => {
  const { getByTestId } = render(
    <BarAlert
      alertText="Warning, something went wrong"
      alertType="danger"
      removeAlert={() => null}
      id="#2345"
    />
  );
  expect(getByTestId("bar-alert")).toHaveTextContent(
    "Warning, something went wrong"
  );
});

it("matches snapshot 1", () => {
  const DOMTree = renderer
    .create(
      <BarAlert
        alertText="Warning, something went wrong"
        alertType="danger"
        removeAlert={() => null}
        id="#2345"
      />
    )
    .toJSON();

  expect(DOMTree).toMatchSnapshot();
});
