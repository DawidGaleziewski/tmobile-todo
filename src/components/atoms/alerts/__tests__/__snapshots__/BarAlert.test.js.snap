// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-5007zv-BarAlert"
  data-testid="bar-alert"
>
  Warning, something went wrong
  <span
    className="css-166am21-BarAlert"
    onClick={[Function]}
  >
    X
  </span>
</div>
`;

exports[`matches snapshot 1 1`] = `
<div
  className="css-5007zv-BarAlert"
  data-testid="bar-alert"
>
  Warning, something went wrong
  <span
    className="css-166am21-BarAlert"
    onClick={[Function]}
  >
    X
  </span>
</div>
`;
